src/main/kotlin/com/zim/pokemon_api/controller/PokemonController.kt

```
@@ -0,0 +1,27 @@
+package com.zim.pokemon_api.controller
+
+import com.zim.pokemon_api.model.Pokemon
+import com.zim.pokemon_api.service.PokemonService
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping("/api/pokemon")
+class PokemonController(private val pokemonService: PokemonService) {
+
+    @GetMapping
+    fun getAllPokemon(): ResponseEntity<List<Pokemon>> {
+        return ResponseEntity.ok(pokemonService.getAllPokemon())
```

Since we're using @RestController annotation here and not @Controller, do we need to wrap everything in ResponseEntity?

---

Good point! In the getAllPokemon method,
there's no need to wrap the method and response with ResponseEntity.
However, in getPokemonById, I needed a simple way to return a 404 status code in case the Pokemon was not found.
Since I hadn't yet implemented a global error handler or thrown a custom error, I used ResponseEntity.notFound()
as a solution to handle that case.
After reconsidering my approach,
I think it will be better to throw a custom error instead and handle it globally in getPokemonById(),
So, there's no longer a need to wrap the getPokemonById response with ResponseEntity either:)